kiosk/src/Models/Kiosk.ts

```diff
@@ -30,27 +40,36 @@ export class Kiosk {
     private builtGamesCache: { [gameId: string]: BuiltSimJSInfo } = { };
     private defaultGameDescription = "Made with love in MakeCode Arcade";
 
-    constructor(clean: boolean, locked: boolean, time?: string) {
+    constructor(opts: KioskOpts) {
+        const { clean, locked, time, shareSrc } = opts;
         this.clean = clean;
-        this.locked = locked;
+        this.shareSrc = shareSrc;
+         // TODO figure out this interaction; right now treating loaded shareSrc as readonly to keep easy
+        this.locked = locked || !!shareSrc;
         this.time = time;
     }
 
     async downloadGameListAsync(): Promise<void> {
+        if (this.shareSrc) {
+            const [id, ...filename] = this.shareSrc.split("/");
+
+            const kioskData = await getSharedKioskData(id, filename.join("/"));
```

Comment:
Is this just for leaving us the option to send the shared kiosk entries to a different file/nested file? If I'm understanding everything correctly, is the filename `kiosk.json` every time?

Reply:
In this one it's emitting as kiosk.json every time yes; I wrote it like this just to keep it simple, but we could choose to omit the filename when sharing it, and default to kiosk.json when filename is empty.

This mirrors our tutorial loading logic from shared projects / github projects -- defaulting to readme.md, but allowing a user to e.g. store multiple kiosks in one project if they wish to, just with a bit more specification involved on which file to read from.